tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -237,10 +233,15 @@ public TracerSettings(IConfigurationSource source)
         /// <summary>
         /// Gets or sets a value indicating whether correlation identifiers are
         /// automatically injected into the logging context.
-        /// Default is <c>false</c>.
+        /// Default is <c>false</c>, unless <see cref="ConfigurationKeys.DirectLogSubmission.EnabledIntegrations"/>
+        /// enables Direct Log Submission.
         /// </summary>
         /// <seealso cref="ConfigurationKeys.LogsInjectionEnabled"/>
-        public bool LogsInjectionEnabled { get; set; }
+        public bool LogsInjectionEnabled
+        {
+            get => LogSubmissionSettings?.LogsInjectionEnabled ?? false;
```

Comment:
This isn't _quite_ right, as it doesn't account for the fact that if LogSubmissionSettings is ultimately enabled, then the default is `true`, not `false`. Unfortunately, it's not easy to know whether direct log submission will be enabled at this point, as we don't decide until we get to `ImmutableTracerSettings` (where we create the final `ImmutableDirectLogSubmissionSettings`). 

I think this is, again, an acceptable trade-off, as it seems unlikely that users are calling the getter here? The alternative is to remove `DirectLogSubmissionSettings` entirely, and create the `ImmutableDirectLogSubmissionSettings` in the `TracerSettings` constructor?

Reply:
This seems perfectly fine to me. The `TracerSettings` object is basically an input into the Tracer configuration so the resulting `ImmutableTracerSettings` is the object that has the true result

Reply:
I believe it's acceptable given the current model (and we should change this model /rant)